vllm/model_executor/models/falcon.py

```diff
@@ -325,7 +321,7 @@ def forward(
             # only one all-reduce operator to reduce the results from
             # both MLP and Attention layers.
             mlp_output += attention_output
-            mlp_output = reduce_from_tensor_model_parallel_region(mlp_output)
+            mlp_output = tenosr_model_parallel_all_reduce(mlp_output)
```

Comment:
Just curios: Can we also somehow hide this into `parallel_utils`?

Reply:
This can be a bit complicated. Here we have two parallel linear and attention layers and we have a small optimization to only use one all-reduce for all their results. I cannot immediately come up with a good way to hide this into `parallel_utils`.

Reply:
Got it. Thanks for the explanation!